Quarkus Jib fails if username contains a space although it is not even used

**Describe the bug**
When adding the jib extension to a freshly created quarkus project (generated with code.quarkus.io), `gradlew assemble` fails with 

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':quarkusBuild'.
> io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.container.image.deployment.ContainerImageProcessor#publishImageInfo threw an exception: java.lang.IllegalArgumentException: The supplied combination of container-image group 'andreas eberle' and name 'code-with-quarkus' is invalid
        at io.quarkus.container.image.deployment.ContainerImageProcessor.publishImageInfo(ContainerImageProcessor.java:62)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:972)
        at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
        at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
        at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
        at java.base/java.lang.Thread.run(Thread.java:834)
        at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```

It seems the jib dependency uses my username (Windows user, hence with a space in the name), as the group for the image as default value. One improvement would be if you replace spaces with a `-` to prevent such issues. 

However, the more important part is that I haven't enabled jib building. I only run `gradlew assemble`. You have to enable `quarkus.container-image.build` or `quarkus.container-image.push` to trigger the jib building. Still the normal `gradlew assemble` fails.

Workaround: Set a value for `quarkus.container-image.group` in `application.properties`. (Normally this property is set by my CI pipeline, when it creates the images, that's why I normally do not have it specified in the `application.properties`.)
[2020-12-02-jib-no-defaults-error-with-space-username.zip](https://github.com/quarkusio/quarkus/files/5629370/2020-12-02-jib-no-defaults-error-with-space-username.zip)



**Expected behavior**
Normal `gradlew assemble` works.

**Actual behavior**
It fails although jib is not activated. 


**To Reproduce**

Steps to reproduce the behavior:
1. Download and unzip 
[2020-12-02-jib-no-defaults-error-with-space-username.zip](https://github.com/quarkusio/quarkus/files/5629376/2020-12-02-jib-no-defaults-error-with-space-username.zip)
2. run `gradlew assemble` if you have a username with a space or set `quarkus.container-image.group=a b` in the `application.properties` file first.

--- a/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java
+++ b/extensions/container-image/deployment/src/main/java/io/quarkus/container/image/deployment/ContainerImageConfig.java
@@ -87,6 +87,7 @@ public class ContainerImageConfig {
     /**
      * Since user.name which is default value can be uppercase and uppercase values are not allowed
      * in the repository part of image references, we need to make the username lowercase.
+     * If spaces exist in the user name, we replace them with the dash character.
      *
      * We purposely don't change the value of an explicitly set group.
      */
@@ -94,7 +95,7 @@ public class ContainerImageConfig {
         if (group.isPresent()) {
             String originalGroup = group.get();
             if (originalGroup.equals(System.getProperty("user.name"))) {
-                return Optional.of(originalGroup.toLowerCase());
+                return Optional.of(originalGroup.toLowerCase().replace(' ', '-'));
             }
         }
         return group;

